Libraries/xcexecution/Sources/NinjaExecutor.cpp

```diff
@@ -660,10 +697,18 @@ buildTargetInvocations(
             }
         }
 
-        /* Write invocations to run after auxiliary files. */
         // TODO(grp): This should perhaps be a separate flag for a 'phony' invocation.
-        if (!invocation.executable().path().empty()) {
-            if (!buildInvocation(&writer, invocation, dependencyInfoToolPath, temporaryDirectory, targetWriteAuxiliaryFiles)) {
+        if (invocation.executable()) {
+            /* Find invocation executable. */
+            ext::optional<std::string> executablePath = NinjaExecutablePath(processContext, filesystem, targetEnvironment.executablePaths(), *invocation.executable());
+            if (!executablePath) {
+                fprintf(stderr, "unable to find executable: %s\n", invocation.executable()->builtin().value_or(invocation.executable()->external().value_or("<NONE>")).c_str());
```

Comment:
That value is only used in errors like this one — otherwise, it uses the resolved `executablePath` in the log. I avoided adding a method to avoid making the warning message value more obvious than the correct output value.